File: app/src/main/java/com/originb/inkwisenote/modules/noterelation/data/NoteRelationDao.java
package com.originb.inkwisenote.modules.noterelation.data;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;
import java.util.Set;

@Dao
public interface NoteRelationDao {
    @Query("SELECT * FROM note_relation ")
    List<NoteRelation> getAllNoteRelations();

    @Query("SELECT * FROM note_relation WHERE note_id = :noteId or related_note_id = :noteId")
    List<NoteRelation> getRelatedNotesOf(Long noteId);

    @Query("SELECT * FROM note_relation WHERE note_id IN (:noteIds) OR related_note_id IN (:noteIds)")
    List<NoteRelation> getRelatedNotesOf(Set<Long> noteIds);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertNoteRelatedNotes(Set<NoteRelation> noteRelation);

    // Alternatively, for custom delete queries, use @Query
    @Query("DELETE FROM note_relation WHERE note_id IN (:noteIds) or related_note_id IN (:noteIds)")
    void deleteByNoteId(List<Long> noteIds);

    // Alternatively, for custom delete queries, use @Query
    @Query("DELETE FROM note_relation WHERE note_id = :noteId or related_note_id = :noteId")
    void deleteByNoteId(Long noteId);

}

File: app/src/main/java/com/originb/inkwisenote2/modules/noterelation/data/NoteRelationDao.kt
package com.originb.inkwisenote2.modules.noterelation.data

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface NoteRelationDao {
    @get:Query("SELECT * FROM note_relation ")
    val allNoteRelations: List<NoteRelation?>?

    @Query("SELECT * FROM note_relation WHERE note_id = :noteId or related_note_id = :noteId")
    fun getRelatedNotesOf(noteId: Long?): List<NoteRelation?>?

    @Query("SELECT * FROM note_relation WHERE note_id IN (:noteIds) OR related_note_id IN (:noteIds)")
    fun getRelatedNotesOf(noteIds: Set<Long>?): List<NoteRelation?>?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertNoteRelatedNotes(noteRelation: Set<NoteRelation?>?)

    // Alternatively, for custom delete queries, use @Query
    @Query("DELETE FROM note_relation WHERE note_id IN (:noteIds) or related_note_id IN (:noteIds)")
    fun deleteByNoteId(noteIds: List<Long>?)

    // Alternatively, for custom delete queries, use @Query
    @Query("DELETE FROM note_relation WHERE note_id = :noteId or related_note_id = :noteId")
    fun deleteByNoteId(noteId: Long?)
}
